slsim/lens_pop.py

```diff
@@ -116,9 +132,15 @@ def __init__(
         if deflector_type in ["elliptical", "all-galaxies"] or source_type in [
             "galaxies"
         ]:
-            pipeline = SkyPyPipeline(
+            pipeline_deflector = SkyPyPipeline(
```

Comment:
I see that in the case of two different sky areas, SkyPy needs to be run twice. Can you make an exception that when the area of the deflector and sources are the same, that this does not happen?